### ISSUE TITLE
Warning useLayoutEffect in SSR

### ISSUE THREAD

**Issue by pecoram** [maslianok/react-resize-detector#138]
Hi,
Using useResizeDetector in ssr with next.js I get this error/warning:
```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
```

Version 6.5.0

**Comment by whplh** [comment]
Hey, 
the useLayoutEffect hook was used to avoid that ResizeDetector will [lag behind](https://github.com/facebook/react/issues/14387#issuecomment-455415008) when the ref changes.
But since this only occurs on ref changes and because SSR seems to be [the future](https://reactjs.org/blog/2020/12/21/data-fetching-with-react-server-components.html) of many react components - @maslianok  Would it actually be too bad to have useEffect again?

**Comment by whplh** [comment]
Other libraries tend to use this "[recommend hack](https://medium.com/@alexandereardon/uselayouteffect-and-ssr-192986cdcf7a)"  [E.g.](https://github.com/mui-org/material-ui/blob/2c8721ccf11740589c975eec46775f3a3ede830d/packages/material-ui/src/utils/useEventCallback.js#L3)

**Comment by maslianok** [comment]
The fix published in v6.6.0
